apm-agent-plugins/apm-quartz-job-plugin/src/test/java/co/elastic/apm/agent/quartz/job/JobTransactionNameInstrumentationTest.java

```diff
@@ -0,0 +1,194 @@
+    }
+    @Test
+    void testJobWithGroup() throws SchedulerException, InterruptedException {
+        JobDetail job = JobBuilder.newJob(TestJob.class)
+    			.withIdentity("dummyJobName", "group1").build();
+        Trigger trigger = TriggerBuilder
+        		.newTrigger()
+        		.withIdentity("myTrigger")
+        		.withSchedule(
+        				SimpleScheduleBuilder.repeatSecondlyForTotalCount(1, 1))
+        		.build();
+        scheduler.scheduleJob(job, trigger);
+        scheduler.start();
+        Thread.sleep(250);
+        scheduler.pauseAll();
+        assertTests(job);
+    }
+
+    @Test
+    void testJobWithoutGroup() throws SchedulerException, InterruptedException {
+        JobDetail job = JobBuilder.newJob(TestJob.class)
+    			.withIdentity("dummyJobName").build();
+        Trigger trigger = TriggerBuilder
+        		.newTrigger()
+        		.withIdentity("myTrigger")
+        		.withSchedule(
+        				SimpleScheduleBuilder.repeatSecondlyForTotalCount(1, 1))
+        		.build();
+        scheduler.scheduleJob(job, trigger);
+        scheduler.start();
+        Thread.sleep(250);
+        scheduler.pauseAll();
+        assertTests(job);
+    }
+    
+    @Test
+	void testJobManualCall() throws SchedulerException, InterruptedException {
+        TestJob job=new TestJob();
+        job.execute(null);
+        assertThat(reporter.getTransactions().size()).isEqualTo(1);
+        assertThat(reporter.getTransactions().get(0).getName()).isEqualToIgnoringCase("TestJob#execute");
+    }
+
+    @Test
+    void testSpringJob() throws SchedulerException, InterruptedException {
+        JobDetail job = JobBuilder.newJob(TestSpringJob.class)
+    			.withIdentity("dummyJobName", "group1").build();
+        Trigger trigger = TriggerBuilder
+        		.newTrigger()
+        		.withIdentity("myTrigger")
+        		.withSchedule(
+        				SimpleScheduleBuilder.repeatSecondlyForTotalCount(1, 1))
+        		.build();
+        scheduler.scheduleJob(job, trigger);
+        scheduler.start();
+        Thread.sleep(250);
+        scheduler.pauseAll();
+        assertTests(job);
+    }
+
+    @Test
+    void testJobWithResult() throws SchedulerException, InterruptedException {
+        JobDetail job = JobBuilder.newJob(TestJobWithResult.class)
+    			.withIdentity("dummyJobName").build();
+        Trigger trigger = TriggerBuilder
+        		.newTrigger()
+        		.withIdentity("myTrigger")
+        		.withSchedule(
+        			SimpleScheduleBuilder.repeatSecondlyForTotalCount(1, 1))
+        		.build();
+        scheduler.scheduleJob(job, trigger);
+        scheduler.start();
+        Thread.sleep(250);
+        scheduler.pauseAll();
+        assertTests(job);
+        assertThat(reporter.getTransactions().get(0).getResult()).isEqualTo("this is the result");
+    }
+    
+    private static AtomicInteger count = new AtomicInteger(0);
```

Comment:
Currently you assert for a single Transaction each test, do we need this counter?